uniffi/src/support/mod.rs

```diff
@@ -194,37 +195,33 @@ unsafe impl<T: Liftable + Lowerable> ViaFfi for Vec<T> {
     }
 }
 
-unsafe impl<'a> ViaFfi for &'a str {
-    type Value = ffi_support::FfiStr<'a>;
-    fn try_from_ffi_value(v: Self::Value) -> Result<Self> {
-        Ok(v.as_str())
-    }
-
-    // This should never happen since there is asymmetry with strings
-    // We typically go FfiStr -> &str for argumetns and
-    // String -> *mut c_char for return values.
-    // We panic for now, the ViaFfi triat will be
-    // broken down to IntoFfi and TryFromFfi to prevent us from having
-    // to add this
-    fn into_ffi_value(&self) -> Self::Value {
-        panic!("Invalid conversion. into_ffi_value should not be called on a &str, a String -> *mut c_char conversion should be used instead.")
-    }
-}
-
+/// Support for passing Strings back and forth across the FFI.
+///
+/// Unlike many other implementations of `ViaFfi`, this passes a pointer rather
+/// than copying the data from one side to the other. This is a safety hazard,
+/// but turns out to be pretty nice for useability.
+///
+/// (In practice, we do end up copying the data, the copying just happens on
+/// the foreign language side rather than here in the rust code.)
```

Comment:
If we don't like the unsafety here, we could implement this logic on top of `RustBuffer` and do a little extra copying. But the `CString` stuff seems sufficiently well-developed that we should be able to manage the safety, similar to how ffi_support manages it internally.